webapp/src/pages/ChatroomPage/ChatroomPage.tsx

```diff
@@ -1,42 +1,104 @@
-import { useEffect } from 'react';
-import { useParams } from 'react-router-dom';
+import { useCallback, useEffect, useState } from 'react';
+import { Navigate, useParams } from 'react-router-dom';
 import { ChatroomMessageInput, ChatroomMessages } from './components';
-import { Header, IconVariant } from '../../shared/components';
+import {
+  Header,
+  IconVariant,
+  Loading,
+  Text,
+  TextColor,
+  TextVariant,
+} from '../../shared/components';
+import { CHATROOM_URL } from '../../shared/urls';
 import socket from '../../shared/socket';
 import './ChatroomPage.css';
 import { useNavigation } from '../../shared/hooks/UseNavigation';
-import { WsException } from '../../shared/types';
+import { SubmitStatus, WsException } from '../../shared/types';
+import { useData } from '../../shared/hooks/UseData';
+import { chatApi } from '../../shared/services/ApiService';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import { User } from '../../shared/generated';
 
-function Chatroom() {
+export default function ChatroomPage() {
   const { chatroomId } = useParams();
+  const { authUser } = useAuth();
+
+  if (!chatroomId || !authUser) {
+    return null;
+  }
+  return <Chatroom chatroomId={chatroomId} authUser={authUser} />;
+}
+
+type ChatroomProps = {
+  chatroomId: string;
+  authUser: User;
+};
+
+function Chatroom({ chatroomId, authUser }: ChatroomProps) {
   const { goBack } = useNavigation();
+  const [status, setStatus] = useState<SubmitStatus>({
+    type: 'pending',
+    message: '',
+  });
+  const getChatroom = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading: isChatroomLoading } = useData(getChatroom);
+  const getChatroomMember = useCallback(
+    () =>
+      chatApi.chatControllerGetChatroomMember({
+        chatroomId,
+        userId: authUser.id,
+      }),
+    [chatroomId, authUser.id],
+  );
+  const { data: chatroomMember, isLoading: isChatroomMemberLoading } =
```

Comment:
Non blocking, but, what do you think about the idea of keeping a local copy of all chatmembers with userId === authuser, and update them when we also update the db, the way we do with useAuth? Here we wouldn't need to call the ChatAPI.

Reply:
Not a bad idea. I can think of some edge cases. For example, the user could browse from multiple browsers. If they join/leave from one browser, we store that info locally, and other browsers won't be aware until they refresh the page. Or an owner/admin could block them, and even if they are members, they can't access the chatroom.
I don't know how we could sync this info.